src/adapters/audienceNetwork.js

```diff
@@ -0,0 +1,208 @@
+</script>
+${nativeContainer}</div></body></html>`;
+};
+
+/**
+ * Creates a "good" Bid object with the given bid ID and CPM.
+ * @param {String} placementId
+ * @param {String} bidId
+ * @param {String} size
+ * @param {Number} cpmCents
+ * @returns {Object} Bid
+ */
+const createSuccessBidResponse = (placementId, size, bidId, cpmCents) => {
+  const bid = createBid(STATUS.GOOD, { bidId });
+  // Prebid attributes
+  bid.bidderCode = getBidderCode();
+  bid.cpm = cpmCents / 100;
+  bid.ad = createAdHtml(placementId, size, bidId);
+  if (!isNative(size)) {
+    [bid.width, bid.height] = size.split('x').map(Number);
+  }
+  // Audience Network attributes
+  bid.hb_bidder = 'fan';
+  bid.fb_bidid = bidId;
+  bid.fb_format = size;
+  bid.fb_placementid = placementId;
+  return bid;
+};
+
+/**
+ * Creates a "no bid" Bid object.
+ * @returns {Object} Bid
+ */
+const createFailureBidResponse = () => {
+  const bid = createBid(STATUS.NO_BID);
+  bid.bidderCode = getBidderCode();
+  return bid;
+};
+
+/**
+ * Fetch bids for given parameters.
+ * @param {Object} bidRequest
+ * @param {Array} params.bids - list of bids
+ * @param {String} params.bids[].placementCode - Prebid placement identifier
+ * @param {Object} params.bids[].params
+ * @param {String} params.bids[].params.placementId - Audience Network placement identifier
+ * @param {Array} params.bids[].sizes - list of accepted advert sizes
+ * @param {Array|String} params.bids[].sizes[] - one of 'native', '300x250', '300x50', [300, 250], [300, 50]
+ * @returns {void}
+ */
+const callBids = bidRequest => {
+  // Build lists of adUnitCodes, placementids and adformats
+  const adUnitCodes = [];
+  const placementids = [];
+  const adformats = [];
+  bidRequest.bids
+    .filter(validateBidRequest)
+    .filter(validateBidRequestSizes)
+    .forEach( bid => bid.sizes.forEach( size => {
+      adUnitCodes.push(bid.placementCode);
+      placementids.push(bid.params.placementId);
+      adformats.push(size);
+    }));
+
+  if (placementids.length) {
+    // Build URL
+    const testmode = isTestmode();
+    const url = format({
+      protocol: 'https',
+      host: 'an.facebook.com',
+      pathname: '/v2/placementbid.json',
+      search: {
+        sdk: '5.5.web',
+        testmode,
+        placementids,
+        adformats
+      }
+    });
+    // Request
+    ajax(url, res => {
+      // Handle response
+      const data = parseJson(res);
+      if (data.errors && data.errors.length) {
+        addBidResponse(null, createFailureBidResponse());
```

Comment:
`addBidResponse` should take an ad unit code as the first param even for `nobid` responses